server/src/main/java/org/elasticsearch/TransportVersions.java

```diff
@@ -142,6 +142,7 @@ static TransportVersion def(int id) {
     public static final TransportVersion NODE_INFO_INDEX_VERSION_ADDED = def(8_500_075);
     public static final TransportVersion FIRST_NEW_ID_LAYOUT = def(8_501_00_0);
     public static final TransportVersion COMMIT_PRIMARY_TERM_GENERATION = def(8_501_00_1);
+    public static final TransportVersion SEARCH_RESPONSE_CLUSTERS_COUNTERS_ADDED = def(8_501_00_2);
```

Comment:
Here is the [PR](https://github.com/elastic/elasticsearch/pull/99731) to add this transport version. Once merged I'll remove this line and therefore the TransportVersions conflict will be resolved.